scripts/ci/libraries/_push_pull_remove_images.sh

```diff
@@ -44,213 +44,22 @@ function push_pull_remove_images::push_image_with_retries() {
 }
 
 
-# Pulls image in case it is needed (either has never been pulled or pulling was forced
+# Pulls image in case it is missing
 # Should be run with set +e
 # Parameters:
 #   $1 -> image to pull
-#   $2 - fallback image
-function push_pull_remove_images::pull_image_if_not_present_or_forced() {
+function push_pull_remove_images::pull_image_if_missing() {
     local image_to_pull="${1}"
     local image_hash
     image_hash=$(docker images -q "${image_to_pull}" 2> /dev/null || true)
-    local pull_image=${FORCE_PULL_IMAGES}
-
     if [[ -z "${image_hash=}" ]]; then
-        pull_image="true"
-    fi
-    if [[ "${pull_image}" == "true" ]]; then
         echo
         echo "Pulling the image ${image_to_pull}"
         echo
         docker pull "${image_to_pull}"
     fi
 }
 
-# Rebuilds python base image from the latest available Python version if it has been updated
```

Comment:
Same here - all that was needed to check and pull the right images before we could use them as cache. This can be removed now completely.